src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -129,16 +132,26 @@ impl LogicalAgg {
             .into(),
         );
         let vnode_col_idx = exprs.len() - 1;
+        // TODO(kwannoel): We should apply Project optimization rules here.
         let project = StreamProject::new(LogicalProject::new(stream_input, exprs));
 
+        // Generate local agg step
         let mut local_group_key = self.group_key().to_vec();
         local_group_key.push(vnode_col_idx);
         let n_local_group_key = local_group_key.len();
         let local_agg = StreamHashAgg::new(
             LogicalAgg::new(self.agg_calls().to_vec(), local_group_key, project.into()),
             Some(vnode_col_idx),
         );
-
+        // Global group key excludes vnode.
+        let local_group_key_without_vnode =
+            &local_agg.group_key()[..local_agg.group_key().len() - 1];
+        let global_group_key = local_agg
+            .i2o_col_mapping()
+            .rewrite_dist_key(local_group_key_without_vnode)
+            .unwrap_or_else(|| panic!("some input group key could not be mapped"));
+
+        // Generate global agg step
```

Comment:
So, in our current e2e tests, all the group key here is [0..group_key().len()]? :open_mouth:

Reply:
Previously `group_key` is `self.group_key()` which was wrong.
Currently e2e tests I wrote don't sufficiently test this case, if at all. But the error was picked up by sqlsmith.
Newly added e2e test cases will test it.